### DISCUSSION TITLE
Simplifying Repository Setup and App Generation for Easy Tracking and Customization?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [PhilDL/remix-gospel-stack#35] [Q&A]
### Simplifying Repository Setup and Tracking Updates

I recently cloned the repository again from the source and started merging my own packages and apps back in while cleaning up unnecessary components, like "fly". During this process, I had an idea to make the repository more user-friendly for tracking updates and adding custom code and packages.

### Current Challenges
Currently, there are a few challenges that need addressing:

Removal of Unnecessary Packages: To start with a clean base, one would have to manually remove existing app packages that are not required. It would be beneficial to have a clean foundation that doesn't implement any packages or apps.
Automating App Generation: It would be helpful to create a CLI script that generates new apps or packages. For example, the CLI could spin up a Remix app with minimal configurations and modify relevant files, such as remix.config, tsconfig, and package.json, to include specific packages.
While I understand that these ideas may extend beyond the current scope of the repository, I wanted to share them as they came to mind.

### Streamlining App and Package Creation
One suggestion is to develop a CLI tool that automates the setup process for new apps or packages. For instance, the CLI could scaffold a new package or Remix app, ensuring that all the necessary configurations are in place.

When manually adding a Remix app currently, several files need modification. With a CLI tool, this process could be automated, making it easier to set up a new app correctly.

### Documentation for App Modification
To further simplify the process, it would be beneficial to create separate documentation outlining the required modifications for a Remix app. This documentation could cover areas such as remix.config, watch patch, package.json, tsconfig, and potentially turbo/pnpm files.

By having clear documentation, users can more easily add apps or packages without overlooking any necessary steps.

I apologize if this post seems a bit chaotic. I wanted to share this idea, and for now, I'm manually making all the modifications and hoping not to make any mistakes. In my current case, I have around five packages and two Remix apps.

Overall, the goal is to simplify the repository setup, streamline app/package generation through a CLI, and provide clear documentation for modifications.

This repro would be a really good showcase but a bare bones repro with some cli that lets you add apps and packages would be more future-proof in terms of cloning and updating the repro

**Comment by Jarrodsz** [comment]
Epic stack has a really nice cli. When you init the project.

Perhaps a simple cli to put all the package files in place with the right config could be a nice addition?

**Reply by PhilDL** [reply]
Hi, I actually did a lot of improvement to the gospel stack cli also, you should try it again:

```sh
pnpm create remix@latest --init-script --install --template https://github.com/PhilDL/remix-gospel-stack
```

You can choose between postgres and sqlite lite-fs, then following the instructions with `pnpm setup` it generates the first migration, and build everything. 

You can find all the code in the `remix.init` folder, if you want to improve it by proposing an option to delete useless packages I would appreciate a PR.
